app/result_generator.py

```diff
@@ -14,6 +14,8 @@ def __init__(self, parser):
         }
 
     async def generate(self, words):
+        if self.parser is None:
```

Comment:
Could go either way here, the ability schema dictates that the `parser` is a list (therefore iterable), but if `self.parser` is null, line 19 will throw a `TypeError` without this check.  From my understanding, `self.parser` can never be null-valued, if this is not the case, we could consider changing line 9 to `self.parser = parser or []`.  Either of these could allow us to remove this check,